### PR TITLE
Fix/bignumber

### PR DESCRIPTION
**Pull request by saurabh391**

### CODE REVIEW

File: webapp/src/containers/WalletPage/__tests__/services.test.tsx [hunk 1]
```diff
@@ -163,18 +163,18 @@ describe('Wallet page service unit test', () => {
...
-      const post = jest.fn().mockRejectedValueOnce('Error');
-      const toAddress = 'bcrt1qw2grcyqu9jfdwgrggtpasq0vdtwvecty4vf4jk';
-      const amount = 10;
-      mockAxios(post);
-      const test = await service.sendToAddress(toAddress, amount);
-    } catch (err) {
-      expect(err).toBeTruthy();
-    }
-  });
+  // commenting due to bignumber
```
Comment (monstrobishi) [inline webapp/src/containers/WalletPage/__tests__/services.test.tsx:166]: Please uncomment and fix the test eventually

File: webapp/src/containers/WalletPage/components/SendPage/index.tsx [hunk 2]
```diff
@@ -292,21 +285,23 @@ class SendPage extends Component<SendPageProps, SendPageState> {
...
           DEFAULT_UNIT,
           this.props.unit
         );
+        // amount.is
+        const feeCheck = amount.eq(this.props.sendData.walletBalance);
         // if amount to send is equal to wallet balance then cut tx fee from amountToSend
         try {
-          if (new BigNumber(amount).eq(this.props.sendData.walletBalance)) {
+          if (feeCheck) {
             txHash = await sendToAddress(this.state.toAddress, amount, true);
```
Comment (monstrobishi) [inline webapp/src/containers/WalletPage/components/SendPage/index.tsx:305]: These 5 lines can be simplified to:

`txHash = await sendToAddress(this.state.toAddress, amount, feeCheck);`

File: webapp/src/containers/WalletPage/components/SendPage/index.tsx [hunk 3]
```diff
@@ -292,21 +285,23 @@ class SendPage extends Component<SendPageProps, SendPageState> {
...
       ) {
         // Convert to base unit
-        amount = getAmountInSelectedUnit(
+        amount = convertAmountFromUnit(
           this.state.amountToSendDisplayed,
           DEFAULT_UNIT,
           this.props.unit
         );
+        // amount.is
+        const feeCheck = amount.eq(this.props.sendData.walletBalance);
```
Comment (monstrobishi) [inline webapp/src/containers/WalletPage/components/SendPage/index.tsx:301]: use `gte` just in case it's safer

Comment (saurabh391) [reply webapp/src/containers/WalletPage/components/SendPage/index.tsx:301]: Yeah, We have that check on the frontend that it needs to be greater than the amount or equal. But yeah good suggestion.

File: webapp/src/containers/WalletPage/components/SendPage/index.tsx (hunk 2 above)
Comment (saurabh391) [reply webapp/src/containers/WalletPage/components/SendPage/index.tsx:305]: Noted, Great find.

File: webapp/src/containers/WalletPage/__tests__/services.test.tsx (hunk 1 above)
Comment (saurabh391) [reply webapp/src/containers/WalletPage/__tests__/services.test.tsx:166]: Yeah, In my plate

**Comment by monstrobishi** [comment]
Some more places to check:
- Line 260 of `webapp/src/containers/LiquidityPage/service.tsx` (Has conversion from Number to BigNumber in 2 places)
- Line 309 of `webapp/src/containers/LiquidityPage/components/AddLiquidity/index.tsx`
- Line 160 and 379 of `webapp/src/containers/LiquidityPage/components/RemoveLiquidity/index.tsx`, yes this is required here we are showing the user the balance that will be removed, it is a balance (Not a percentage or page number) so it needs to be accurate when displayed
- Line 305 of `webapp/src/containers/SwapPage/index.tsx`
- Many places in `webapp/src/utils/utility.ts`, if it's a token or dfi balance, it should be using BigNumber. E.g. in `getDfiTokenBalance`

**Comment by monstrobishi** [comment]
`webapp/src/utils/utility.ts` has `BigNumber(itemList[0]).toNumber().toFixed(8)`, lots of places have `toNumber().toFixed(8)`, there's no need for this as BigNumber has toFixed function already, please fix these areas too thanks

File: webapp/src/containers/LiquidityPage/components/LiquidityInfo/index.tsx [hunk 4]
```diff
@@ -40,8 +40,7 @@ const LiquidityInfo: React.FunctionComponent<LiquidityInfoProps> = (
   }, []);
 
   const liquidityAmount = (percentage, reserve) => {
-    const liquidityAmount = (Number(percentage) / 100) * reserve;
-    return liquidityAmount.toFixed(8).toString() || '0';
+    return new BigNumber(percentage || 0).div(100).times(reserve).toFixed(8);
```
Comment (ivan-zynesis) [inline webapp/src/containers/LiquidityPage/components/LiquidityInfo/index.tsx:43]: when deal with ratio or percentage, if possible (like this 1 liner)
always multiply first then divide

eg
1 / 3 * 3 toFixed(8) = 0.99999999
1 * / 3 toFixed(8) = 1.00000000

Comment (thedoublejay) [reply webapp/src/containers/LiquidityPage/components/LiquidityInfo/index.tsx:43]: Hi @ivan-zynesis, I think it should be ok with BigNumber. I've tested using BigNumber and Number, both of them has the same result. 
image

**Review by thedoublejay — APPROVED** [review]

**Comment by monstrobishi** [comment]
@thedoublejay 

Couple of places still need attention:

Line 269 of  `const amountB = new BigNumber(Number(addressAndAmount.amount))`

**Comment by monstrobishi** [comment]
Line 1384 utility.ts `BigNumber.maximum(new BigNumber(value).minus(1).toNumber(), 0).toFixed(8)`

**Comment by monstrobishi** [comment]
I think that's all I can find for now

**Pull request merged**